src/contentEncoding.ts

```diff
@@ -0,0 +1,166 @@
+/**
+ * Content Encoding Types
+ *
+ * Provides branded types for compile-time safety when handling encoded vs plain text content.
+ * Prevents accidentally mixing base64-encoded and plain text strings.
+ */
+
+/**
+ * Plain text content (UTF-8 string)
+ * Used for text files like .md, .txt when read from local vault
+ */
+export type PlainTextContent = string & { readonly __brand: 'PlainText' };
```

Comment:
For reference, this is a branded type used for static typechecks. It helps typechecks ensure the usage of strings, so anywhere the code passes something like a base64-encoded string where PlainTextContent is expected it should trigger a type mismatch.